File: 54주차/부분삼각수열.js
const solution = () => {
  arr.sort((a, b) => a - b);
  let answer = 0;
  if (arr.length <= 2) console.log(arr.length);
  else {
    for (let i = 0; i < n - 1; i++) {
      for (let j = n - 1; j >= 0; j--) {
        if (arr[i] + arr[i + 1] > arr[j]) {
          answer = Math.max(answer, j - i + 1);
        }
      }
    }
    console.log(answer);
  }
};

const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let n = null;
let arr = [];
let count = 0;

rl.on("line", (line) => {
  if (!n) {
    n = +line;
  } else {
    arr = line.split(" ").map((e) => +e);
    rl.close();
  }
}).on("close", () => {
  solution();
  process.exit();
});
// 백준 문제
// x>y>z이고 x+y>z를 만족하면 삼각관계에 해당함.
// 범위를 늘려가며, 가장 큰 수 부터 검사해서 최댓값 구하기.

File: 63주차/부분삼각수열.py
n = int(input())
arr = list(map(int,input().split()))
answer =0
arr.sort()
if n <=2:
    print(n)
else:
    for i in range(n-1):
        x,y = arr[i],arr[i+1]
        for j in range(n):
            z = arr[n-1-j]
            if x+y > z:
                answer = max(answer, n-1-j-i+1)

    print(answer)
# 정렬한 상태에서 제일 작은 x,y 두 수의 합이 제일 큰 z보다 크다면 삼각관계를 만족한다.
# for 중복을 통해 조건에 만족하는 지점의 수열 길이를 업데이트함.